components/Chips/src/MDCChipView.m

```diff
@@ -450,49 +450,34 @@ - (void)setContentHorizontalAlignment:(UIControlContentHorizontalAlignment)align
 }
 
 - (void)updateTitleFont {
-  UIFont *customTitleFont = _titleFont;
-
   // If we have a custom font apply it to the label.
   // If not, fall back to the Material specified font.
-  if (customTitleFont) {
-    // If we are automatically adjusting for Dynamic Type resize the font based on the text style
-    if (_mdc_adjustsFontForContentSizeCategory) {
-      self.titleLabel.font = [customTitleFont
-          mdc_fontSizedForMaterialTextStyle:kTitleTextStyle
-                       scaledForDynamicType:_mdc_adjustsFontForContentSizeCategory];
-    } else {
-      self.titleLabel.font = customTitleFont;
-    }
-  } else {
-    // TODO(#2709): Migrate to a single source of truth for fonts
-    // There is no custom font, so use the default font.
-    if (_mdc_adjustsFontForContentSizeCategory) {
-      // If we are using the default (system) font loader, retrieve the
-      // font from the UIFont preferredFont API.
-      if ([MDCTypography.fontLoader isKindOfClass:[MDCSystemFontLoader class]]) {
-        _titleLabel.font = [UIFont mdc_preferredFontForMaterialTextStyle:kTitleTextStyle];
-      } else {
-        // There is a custom font loader, retrieve the font and scale it.
-        UIFont *customTypographyFont = [MDCTypography buttonFont];
-        _titleLabel.font = [customTypographyFont
-            mdc_fontSizedForMaterialTextStyle:kTitleTextStyle
-                         scaledForDynamicType:_mdc_adjustsFontForContentSizeCategory];
-      }
+  UIFont *titleFont = _titleFont ?: [[self class] defaultTitleFont];
+
+  // If we are automatically adjusting for Dynamic Type resize the font based on the text style
+  if (self.mdc_adjustsFontForContentSizeCategory) {
+    if (titleFont.mdc_scalingCurve && !self.mdc_legacyFontScaling) {
+      titleFont = [titleFont mdc_scaledFontForCurrentSizeCategory];
     } else {
-      // If we are using the default (system) font loader, retrieve the
-      // font from the UIFont standardFont API.
-      if ([MDCTypography.fontLoader isKindOfClass:[MDCSystemFontLoader class]]) {
-        _titleLabel.font = [UIFont mdc_standardFontForMaterialTextStyle:kTitleTextStyle];
-      } else {
-        // There is a custom font loader, retrieve the font from it.
-        _titleLabel.font = [MDCTypography buttonFont];
-      }
+      titleFont =
+          [titleFont mdc_fontSizedForMaterialTextStyle:kTitleTextStyle
+                                  scaledForDynamicType:_mdc_adjustsFontForContentSizeCategory];
     }
   }
+  self.titleLabel.font = titleFont;
 
   [self setNeedsLayout];
 }
 
++ (UIFont *)defaultTitleFont {
+  // TODO(#2709): Migrate to a single source of truth for fonts
```

Comment:
is this todo still valid?

Reply:
It is still valid after I checking the implementation. Though it is an old one:(